src/lib_ccx/ocr.c

```diff
@@ -97,36 +97,22 @@ void delete_ocr(void **arg)
 char *probe_tessdata_location(const char *lang)
 {
 	int ret = 0;
-	char *tessdata_dir_path = getenv("TESSDATA_PREFIX");
-
-	ret = search_language_pack(tessdata_dir_path, lang);
-	if (!ret)
-		return tessdata_dir_path;
-
-	tessdata_dir_path = "./";
-	ret = search_language_pack(tessdata_dir_path, lang);
-	if (!ret)
-		return tessdata_dir_path;
-
-	tessdata_dir_path = "/usr/share/";
-	ret = search_language_pack(tessdata_dir_path, lang);
-	if (!ret)
-		return tessdata_dir_path;
-
-	tessdata_dir_path = "/usr/local/share/";
-	ret = search_language_pack(tessdata_dir_path, lang);
-	if (!ret)
-		return tessdata_dir_path;
-
-	tessdata_dir_path = "/usr/share/tesseract-ocr/";
-	ret = search_language_pack(tessdata_dir_path, lang);
-	if (!ret)
-		return tessdata_dir_path;
-
-	tessdata_dir_path = "/usr/share/tesseract-ocr/4.00/";
-	ret = search_language_pack(tessdata_dir_path, lang);
-	if (!ret)
-		return tessdata_dir_path;
+
+	const char *paths[] = {
+	    getenv("TESSDATA_PREFIX"),
+	    "./",
+	    "/usr/share/",
+	    "/usr/local/share/",
+	    "/usr/share/tesseract-ocr/",
+	    "/usr/share/tesseract-ocr/4.00/",
+	    "/usr/share/tesseract-ocr/5/",
+	    "/usr/share/tesseract/"};
+
+	for (int i = 0; i < sizeof(paths) / sizeof(paths[0]); i++)
```

Comment:
Consider just adding a NULL at the end of the array and checking for it instead of doing some math to find out the number of elements :-)

Reply:
That would also work but this one is a standard in c to find out the number of elements and it will never get wrong.

Reply:
The first item in the list is also a characters of string, so if the environment variable is empty then it's value will be '\0' which is NULL in C, so this approach is better.

Reply:
Not sure I like having a possible NULL in the middle of a list of pointers :-) But since you did this work and it seems reasonable, your choice.